pkgs/development/libraries/arcan/default.nix

```
@@ -0,0 +1,185 @@
+, espeak-classic
+
+#
+, withLibmagic ? true
+, file
+
+#
+, withTesseractAndLeptonica ? true
+, tesseract
+, leptonica
+
+#
+, withVNC ? true
+, libvncserver
+
+# Output debug information
+, debugging ? false
+}:
+
+let
+  name = "arcan";
+
+  srcs = {
+    arcan = fetchFromGitHub {
+      owner = "letoram";
+      repo = name;
+      rev = "ffc266523bc89def40666cceb17be5707994d1c0";
+      sha256 = "sha256-bfLGjnnxX+sHnDn5eGYKqRR51Gn9al9GEqw9qF63j38=";
+    };
+
+    openal = fetchFromGitHub {
+      owner = "letoram";
+      repo = "openal";
+      rev = "1c7302c580964fee9ee9e1d89ff56d24f934bdef"; # master
+      sha256 = "0dcxcnqjkyyqdr2yk84mprvkncy5g172kfs6vc4zrkklsbkr8yi2";
+    };
+  };
+in stdenv.mkDerivation rec {
+  pname = name;
+
+  version = "0.6.1pre1+unstable=2021-05-31";
+
+  src = srcs.arcan;
+
+  postUnpack = ''
+    mkdir -p $sourceRoot/external/git/openal
+    pushd $sourceRoot/external/git/
+    cp --recursive --verbose ${srcs.openal}/* openal/
+    chmod --recursive 777 openal/
+    popd
+
+    # Generate manpages
+    pushd $sourceRoot/doc/
+    mkdir mantmp
+    ruby docgen.rb mangen
+    popd
+  '';
+
+  postPatch = ''
+    substituteInPlace ./src/platform/posix/paths.c \
+      --replace "/usr/bin" "$out/bin" \
+      --replace "/usr/share" "$out/share"
+
+    substituteInPlace ./src/CMakeLists.txt \
+      --replace "SETUID" ""
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    ruby
+  ];
+
+  buildInputs = [
+    openal
+    freetype
+    sqlite
+    libuvc
+    libGL
+    SDL2
+    libdrm
+    libxkbcommon
+    mesa
+    valgrind
+    luajit
+    libusb1
+    libvlc
+    xz
+    libffi
+    libX11
+    libxcb
+    libXcomposite
+    libXfixes
+    xcbutil
+    xcbutilwm
+    libXau
+    libXdmcp
+    harfbuzzFull
+  ] ++ lib.optionals withFFmpeg [
+    ffmpeg-full
+  ] ++ lib.optionals withWayland [
+    wayland
+    wayland-protocols
+  ] ++ lib.optionals withEspeak [
+    espeak-classic
+  ] ++ lib.optionals withLibmagic [
+    file
+  ] ++ lib.optionals withTesseractAndLeptonica [
+    tesseract
+    leptonica
+  ] ++ lib.optionals withVNC [
+    libvncserver
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_PRESET=everything"
+    "-DISTR_TAG=Nixpkgs"
+    "-DENGINE_BUILDTAG=${version}+Nixpkgs"
+    "../src"
+  ] ++ lib.optionals debugging [ "-DCMAKE_BUILD_TYPE=Debug" ];
+
+  hardeningDisable = [ "format" ] ++ lib.optionals debugging [ "fortify" ];
+
+  meta = with lib; {
+    description = "Scriptable Multimedia Engine";
+    longDescription = ''
+      Arcan  is a portable and fast self-sufficient multimedia engine for
+      advanced visualization and analysis work in a wide range of applications
+      e.g. game development, real-time streaming video,  monitoring  and
+      surveillance, up to and including desktop compositors and window managers.
+    '';
+    homepage = "https://github.com/letoram/${pname}";
+    changelog = "https://github.com/letoram/${pname}/releases/tag/${version}";
```

```suggestion
    homepage = "https://github.com/letoram/arcan";
    changelog = "https://github.com/letoram/arcan/releases/tag/${version}";
```

My browser is not capable to parse nix files to give me a link to the right homepage.